### PR TITLE
[webview_flutter_android] Fix Android lint warnings

### PR DESCRIPTION
**Pull request by bparrishMines**
Part of https://github.com/flutter/flutter/issues/88011

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/CookieManagerHostApiImpl.java [hunk 1]
```diff
@@ -30,7 +32,7 @@ public void setCookie(String url, String value) {
    * @param cookieManager The cookie manager to clear all cookies from.
    * @return Whether any cookies were removed.
    */
-  @SuppressWarnings("deprecation")
+  @SuppressWarnings({"deprecation", "RedundantSuppression"})
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/CookieManagerHostApiImpl.java:35]: This is the scenario where where the `javac` linter and the Android gradle linter disagreed. The `javac` linter claims this is deprecated, but the Android lint believes this does not need a suppression. 

After I did some investigation, It looks like the problem is that the `javac` linter doesn't understand the `if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)` or `@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)` logic. They're other instance of this throughout this plugin, and will probably show up in other plugins as well.

Potentially we should ignore the deprecation warning from the `javac` linter and allow the Android gradle linter to handle it?

@reidbaker FYI

Comment (stuartmorgan) [reply packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/CookieManagerHostApiImpl.java:35]: If the [Android lint](http://tools.android.com/tips/lint-checks) deprecation option covers everything the `javac` lint one does, but automatically handles the version checks, disabling deprecation for `javac` in all plugins sounds good to me. The fact that the javac version made us suppress everything even when we were already doing checks was really annoying.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks for all the cleanup!

> * [ ]  I added new tests to check the change I am making, or this PR is [test-exempt](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests).

The PR should also update (or remove if empty) the `lint-baseline.xml` file, which would serve as a test for this.

File: packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/CookieManagerHostApiImpl.java (hunk 1 above)
Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/CookieManagerHostApiImpl.java]: Ah, I tried it again in Android Studio and I think you're right. I must have got the error mixed up with another one. The lint command does still pass if only `@SuppressWarnings("deprecation")` is used. I should be able to change my local AS settings to ignore this warning.

File: packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/GeneratedAndroidWebView.java [hunk 2]
```diff
@@ -22,15 +22,39 @@
 import java.util.Map;
 
 /** Generated class from Pigeon. */
-@SuppressWarnings({"unused", "unchecked", "CodeBlock2Expr", "RedundantSuppression"})
+@SuppressWarnings({"unused", "unchecked", "CodeBlock2Expr", "RedundantSuppression", "serial"})
```
Comment (reidbaker) [inline packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/GeneratedAndroidWebView.java:25]: For my own education did this suppression come from running pidgen?

**Review by reidbaker — APPROVED** [review]

File: packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/GeneratedAndroidWebView.java (hunk 2 above)
Comment (stuartmorgan) [reply packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/GeneratedAndroidWebView.java:25]: Yes, we added this to the Pigeon generator for Java recently.

File: packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/DisplayListenerProxy.java [hunk 3]
```diff
@@ -118,9 +116,11 @@ private static ArrayList<DisplayListener> yoinkDisplayListeners(DisplayManager d
       return new ArrayList<>();
     }
     try {
+      //noinspection JavaReflectionMemberAccess
```
[the remark this answers is not in the document]
Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/DisplayListenerProxy.java]: Yea, I didn't realize this was unique to AS. I went ahead and removed them

**Comment by bparrishMines** [comment]
@stuartmorgan I removed the baseline and finished fixing the rest of the lint errors

**Review by stuartmorgan — APPROVED** [review]
LGTM!

**Pull request merged**